src/plugins/data/common/search/aggs/buckets/create_filter/ip_prefix.ts

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { buildRangeFilter } from '@kbn/es-query';
+import { IBucketAggConfig } from '../bucket_agg_type';
+
+export const createFilterIpPrefix = (aggConfig: IBucketAggConfig) => {
+  return buildRangeFilter(
```

Comment:
This doesn't seem right to me. Why rangeFilter here? I think this is causing the bugs with the filters creation. It would be awesome if we could add a test too!

Reply:
Updated to use a proper Range Filter that converts the input into a CidrMask (like the ip_range filter) to resolve the behavior issue in your comment above